expertise/service/expertise.py

```diff
@@ -208,128 +108,41 @@ def _get_score_and_metadata_dir(self, search_dir):
-                    index = json.load(f)
-            else:
-                raise OpenReviewException('Bad request: no jobs have been submitted yet')
-
-            # Return the entire list of job IDs
-            if user_id in index.keys():
-                return index[user_id]
-            else:
-                raise OpenReviewException('User not found: no jobs submitted with this user ID')
-    
-    def _del_from_user_index(self, user_id, job_id):
-        """
-        Removes a job ID from the record
-
-        :param user_id: The user ID that the job was submitted by
-        :type user_id: str
-
-        :param job_id: The ID of the job to be added to the record
-        :type job_id: str
-        """
-        # Load existing index, otherwise throw an error
-        with user_index_file_lock:
-            index_path = os.path.join(self.working_dir, 'index.json')
-            if os.path.isfile(index_path):
-                with open(os.path.join(self.working_dir, 'index.json'), 'r') as f:
-                    index = json.load(f)
-            else:
-                raise OpenReviewException('Bad request: no jobs have been submitted yet')
-
-            # Remove the job ID from the list
-            if user_id in index.keys():
-                index[user_id].remove(job_id)
-            else:
-                raise OpenReviewException('User not found: no jobs submitted with this user ID')
-        
-            # Write out the index
-            with open(os.path.join(self.working_dir, 'index.json'), 'w+') as f:
-                json.dump(index, f, ensure_ascii=False, indent=4)
-
     def start_expertise(self, request):
         descriptions = JobDescription.VALS.value
-        job_id = shortuuid.ShortUUID().random(length=5)
-        request['job_id'] = job_id
 
         from .celery_tasks import run_userpaper
-        config = self._prepare_config(request)
+        config, token = self._prepare_config(request)
+        job_id = config.job_id
 
-        self.logger.info(f'Config: {config}')
-        config['mdate'] = int(time.time() * 1000)
-        config['status'] = JobStatus.QUEUED
-        config['description'] = descriptions[JobStatus.QUEUED]
+        config.mdate = int(time.time() * 1000)
+        config.status = JobStatus.QUEUED
+        config.description = descriptions[JobStatus.QUEUED]
 
         # Config has passed validation - add it to the user index
-        self._add_to_user_index(config['user_id'], config['job_id'])
         run_userpaper.apply_async(
-            (config, self.logger),
+            (config.to_json(), token, self.logger),
             queue='userpaper',
             task_id=job_id
         )
-        with open(os.path.join(config['job_dir'], 'config.json'), 'w+') as f:
-            json.dump(config, f, ensure_ascii=False, indent=4)
+        self.logger.info(f"\nconf: {config.to_json()}\n")
+        with open(os.path.join(config.job_dir, 'config.json'), 'w+') as f:
```

Comment:
can you add a method `save_config` in JobConfig class? it seems you are saving the file in different places.